pinot-core/src/main/java/org/apache/pinot/core/segment/processing/genericrow/GenericRowDeserializer.java

```diff
@@ -99,6 +100,15 @@ public void deserialize(long offset, GenericRow buffer) {
             buffer.putValue(fieldName, bytes);
             break;
           }
+          case BIG_DECIMAL: {
```

Comment:
(minor) Put it after `DOUBLE` (same order as data type definition) for easier tracking. Same for other places